### PR TITLE
Add config to use publicURL or internalURL in RackspaceConnector

### PR DESCRIPTION
**Pull request by arosenfeld-mentel**
This commit allows the Rackspace adaptor to use the internal url (sometimes called ServiceNet), which does not charge for bandwidth inside Rackspace servers.

It still defaults to the public url, which means this doesn't change anything unless a config is added. The config can be added as a boolean called "internal" or as a string called "urlType". The reason for having both is to not leak a implementation detail from Rackspace's OpenCloud, which is that urlType must be either internalURL or publicURL, so the internal boolean flag can be used. But we still allow a new urlType to be added in the future.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please add tests and fix the existing tests.

File: src/Adapters/RackspaceConnector.php [hunk 1]
```diff
@@ -64,6 +64,11 @@ protected function getAuth(array $config)
             throw new InvalidArgumentException('The rackspace connector requires a container.');
         }
 
+        $config['urlType'] = array_get($config, 'urlType', array_get($config, 'internal', false) ? 'internalURL' : 'publicURL');
```
Comment (GrahamCampbell) [inline src/Adapters/RackspaceConnector.php]: Is this not a bit overkill here?

**Comment by arosenfeld-mentel** [comment]
Tests added and existing ones should pass, however I don't know how to re-run the Travis CI tests.

File: src/Adapters/RackspaceConnector.php (hunk 1 above)
Comment (arosenfeld-mentel) [inline src/Adapters/RackspaceConnector.php]: Probably. Didn't know to use a flag or a string.

**Comment by GrahamCampbell** [comment]
Merge conflicts?

File: src/config/config.php [hunk 2]
```diff
@@ -128,6 +128,7 @@
             'username'   => 'your-username',
             'apiKey'     => 'your-api-key',
             'container'  => 'your-container',
+            // 'internal'    => true,
```
Comment (GrahamCampbell) [inline src/config/config.php]: Not aligned correctly.

**Comment by GrahamCampbell** [comment]
You haven't got any tests with "Internal"?

**Comment by arosenfeld-mentel** [comment]
Merge conflicts might be coming from the fact that we are still on v1.0, since we still need Laravel 4 support. But GitHub is not showing me what those conflicts are.

**Comment by GrahamCampbell** [comment]
I can't merge this then. You've sent this PR to the 3.0 version.

**Comment by GrahamCampbell** [comment]
Only the 3.0 version is maintained due to the fact that only Laravel 5.1 is actually maintained. Laravel 4 is not maintained anymore.

**Comment by arosenfeld-mentel** [comment]
Didn't know how to change this PR, so I added a new one at #58

**Pull request closed without merging**